Sources/SwiftGraphQLParser/GraphQLParser.swift

```diff
@@ -67,25 +128,27 @@ public func parse(_ input: String) throws -> Document {
 private extension ArraySlice where Element == Token {
 	
 	mutating func readDefinition() throws -> ExecutableDefinition? {
-		if let operationDefinition = try self.readOperationDefinition() {
+		guard let first = self.first else { return nil }
+		
+		if let operationDefinition = try self.readOperationDefinition(startIndex: first.range.lowerBound) {
```

Comment:
tbh I may just be misunderstanding because its been a while since I've seen this code, but do we need to pass the startIndex through to all of these methods? Can the startIndex value just be computed from the `start` variable that is defined in methods like `readFragmentDefinition` etc?

Reply:
I'm not sure we can do that as these methods are `mutating func`s and as such `first` in the initial call may not be the same `first` in the next call (for example, if the parsing succeeds).